xds/internal/balancer/clusterresolver/balancerconfigbuilder/configbuilder.go

```diff
@@ -0,0 +1,308 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package balancerconfigbuilder contains utility functions to build balancer
+// config. The built config will generate a tree of balancers with priority,
+// cluster_impl, weighted_target, lrs, and roundrobin.
+//
+// This is in a subpackage of cluster_resolver so that it can be used by the EDS
+// balancer. Eventually we will delete the EDS balancer, and replace it with
+// cluster_resolver, then we can move the functions to package cluster_resolver,
+// and unexport them.
+//
+// TODO: move and unexport. Read above.
+package balancerconfigbuilder
```

Comment:
I renamed the package to `balancerconfig`.

I didn't change the type names more, because the way they are now would make it easier when we move them to the cluster_resolver package and unexport.